magento/devdocs#:5185  Adding more information about configuring TinyMCE (#6681)
magento/devdocs: src/guides/v2.3/ui_comp_guide/components/wysiwyg/configure-tinymce-editor.md

```xml
<type name="Magento\Cms\Model\Wysiwyg\CompositeConfigProvider">
    <arguments>
        <argument name="variablePluginConfigProvider" xsi:type="array">
            <item name="Magento_PageBuilder/pageBuilderAdapter" xsi:type="string">Magento\Cms\Model\WysiwygDefaultConfig</item>
        </argument>
        <argument name="widgetPluginConfigProvider" xsi:type="array">
            <item name="Magento_PageBuilder/pageBuilderAdapter" xsi:type="string">Magento\Cms\Model\WysiwygDefaultConfig</item>
        </argument>
        <argument name="wysiwygConfigPostProcessor" xsi:type="array">
            <item name="Magento_PageBuilder/pageBuilderAdapter" xsi:type="string">Magento\Cms\Model\WysiwygDefaultConfig</item>
            <item name="default" xsi:type="string">Magento\PageBuilder\Model\Wysiwyg\DefaultConfigProvider</item>
        </argument>
        <argument name="galleryConfigProvider" xsi:type="array">
            <item name="Magento_PageBuilder/pageBuilderAdapter" xsi:type="string">Magento\Cms\Model\WysiwygDefaultConfig</item>
        </argument>
    </arguments>
</type>
```

The class DefaultConfigProvider returns data required to render TinyMCE editor.

> From the example the class `Magento\PageBuilder\Model\Wysiwyg\DefaultConfigProvider` overrides existing configuration provided from the cms module.

Reviewer: ```suggestion
From the example the class `Magento\PageBuilder\Model\Wysiwyg\DefaultConfigProvider` overrides existing configuration provided by `Magento_CMS` module.
```

  </arguments>
</type>
```

Once you have edited the `di.xml` file you can apply custom styling for the settings you implemented in the related CSS file.

> Also, the config of the TinyMCE editor builds can be extended by a plugin.

Reviewer: ```suggestion
Also, the config of the TinyMCE editor can be extended by a plugin.
```

Configuration providers are classes with a `getConfig()` method that returns the configuration for a specific entity.

> For example, in the `Magento\Cms\Model\Wysiwyg\DefaultConfigProvider` the method `getConfig`

Reviewer: ```suggestion
For example, in the `Magento\Cms\Model\Wysiwyg\DefaultConfigProvider` the method `getConfig()`
```

{
    ...

    /**
     * {@inheritdoc}
     */
    public function getConfig(\Magento\Framework\DataObject $config) : \Magento\Framework\DataObject
    {
        $config->addData([
            'tinymce4' => [
                'toolbar' => 'formatselect | bold italic underline | alignleft aligncenter alignright | '
                    . 'bullist numlist | link table charmap',
                'plugins' => implode(
                    ' ',
                    [
                        'advlist',
                        'autolink',
                        'lists',
                        'link',
                        'charmap',
                        'media',
                        'noneditable',
                        'table',
                        'contextmenu',
                        'paste',
                        'code',
                        'help',
                        'table'
                    ]
                ),
                'content_css' => $this->assetRepo->getUrl('mage/adminhtml/wysiwyg/tiny_mce/themes/ui.css')
            ]
        ]);
        return $config;
    }
}
```

> In the case the `Magento\Cms\Model\Wysiwyg\Gallery\DefaultConfigProvider` enables the image plugin.

Reviewer: ```suggestion
In this case, the `Magento\Cms\Model\Wysiwyg\Gallery\DefaultConfigProvider` enables the image plugin.
```

See a list of possibly TinyMCE settings on their [website](https://www.tinymce.com/docs/configure/).

> In this case you can extend and fully customize the TinyMCE editor implementing `DefaultConfigProvider`, the class implements the method `getConfig` where specify necessary configuration. The method can by extended/overwrote of the after plugin.

Reviewer: ```suggestion
The following example shows how you can extend the TinyMCE editor configuration by creating a plugin for `DefaultConfigProvider` which extends  `getConfig()` method of the original class. 
```
